aixplain/model_interfaces/interfaces/function_models.py

```diff
@@ -179,3 +194,207 @@ def predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Di
+            text_generation_dict = text_generation_output["predictions"][i].dict()
+            TextGenerationOutput(**text_generation_dict)
+            text_generation_output["predictions"][i] = text_generation_dict
+        return text_generation_output
+    
+    def _run_tokenize(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
+        instances = request['instances']
+        tokenization_input_list = []
+        # Convert JSON serializables into TextListInputs
+        for instance in instances:
+            tokenization_input = TextListInput(**instance)
+            tokenization_input_list.append(tokenization_input)
+            
+        tokenizer_outputs = []
+        for i in range(len(tokenization_input_list)):
+            tokenization_input = tokenization_input_list[i]
+            token_counts = self.tokenize(tokenization_input)
+            tokenizer_outputs.append(token_counts)
+        
+        tokenizer_output_dict = {
+                "token_counts": tokenizer_outputs
+        }
+        return tokenizer_output_dict
+    
+    def tokenize(self, messages: TextListInput) -> List[int]:
+        pass
+    
+class TextGenerationChatModel(TextGenerationModel):
+    def run_model(self, api_input: Dict[str, List[TextInput]], headers: Dict[str, str] = None) -> Dict[str, List[TextGenerationOutput]]:
+        pass
+
+    def _route(self, request):
+        if "function" in request.keys():
+            function_type = request["function"].upper()
+            if function_type == "PREDICT":
+                return self._predict
+            elif function_type == "TOKENIZE":
+                return self._run_tokenize
+            elif function_type == "TEMPLATIZE":
+                return self._templatize
+            else:
+                raise ValueError("Invalid function.")
+        else:
+            return self._predict
+    
+    def _predict(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
+        instances = request['instances']
+        text_generation_input_list = []
+        # Convert JSON serializables into TextInputs
+        # NOTE: These TextInputs should contain templatized data, which should
+        # have been processed by TEMPLATIZE.
+        for instance in instances:
+            text_generation_input = TextInput(**instance)
+            text_generation_input_list.append(text_generation_input)
+
+        text_generation_output = self.run_model({"instances": text_generation_input_list})
+
+        # Convert JSON serializables into TextGenerationOutputs
+        for i in range(len(text_generation_output["predictions"])):
+            text_generation_dict = text_generation_output["predictions"][i].dict()
+            TextGenerationOutput(**text_generation_dict)
+            text_generation_output["predictions"][i] = text_generation_dict
+        return text_generation_output
+
+    def _templatize(self, request: Dict[str, str], headers: Dict[str, str] = None) -> Dict:
```

Comment:
There should be input and output validation in templatize method. You can remove the underscore from the method name as it's expected for the user to override this.